rtgui/locallabtools2.cc

```diff
@@ -4723,3 +4723,740 @@ void LocallabLog::updateLogGUI()
+
+        // Set default value for adjuster widgets
+        sensimask->setDefault((double)defSpot.sensimask);
+        strumaskmask->setDefault(defSpot.strumaskmask);
+        toolmask->set_active(defSpot.toolmask);
+        blendmask->setDefault((double)defSpot.blendmask);
+        softradiusmask->setDefault((double)defSpot.softradiusmask);
+        radmask->setDefault(defSpot.radmask);
+        lapmask->setDefault(defSpot.lapmask);
+        chromask->setDefault(defSpot.chromask);
+        lapmask->setDefault(defSpot.lapmask);
+        slopmask->setDefault(defSpot.slopmask);
+        shadmask->setDefault(defSpot.shadmask);
+        str_mask->setDefault(defSpot.str_mask);
+        ang_mask->setDefault(defSpot.ang_mask);
+        HHhmask_shape->setCurve(defSpot.HHhmask_curve);
+        contmask->setDefault(defSpot.contmask);
+        blurmask->setDefault(defSpot.blurmask);
+        csThresholdmask->setDefault<int>(defSpot.csthresholdmask);
+
+    }
+
+    // Note: No need to manage pedited as batch mode is deactivated for Locallab
+}
+
+void LocallabMask::updateGUIToMode(const modeType new_type)
+{
+    if (new_type == Normal) {
+        // Advanced widgets are hidden in Normal mode
+        lapmask->hide();
+        gammask->hide();
+        slopmask->hide();
+        shadmask->hide();
+        str_mask->hide();
+        ang_mask->hide();
+        struFrame->hide();
+        blurFrame->hide();
+        gradFramemask->hide();
+        mask_HCurveEditorG->hide();
+//        mask2CurveEditorG->hide();
+        mask2CurveEditorGwav->hide();
+        csThresholdmask->hide();
+
+    } else {
+        // Advanced widgets are shown in Expert mode
+        lapmask->show();
+        gammask->show();
+        slopmask->show();
+        shadmask->show();
+        str_mask->show();
+        ang_mask->show();
+        struFrame->show();
+        blurFrame->show();
+        gradFramemask->show();
+        mask_HCurveEditorG->show();
+//        mask2CurveEditorG->show();
+        mask2CurveEditorGwav->show();
+        csThresholdmask->show();
+
+    }
+}
+
+void LocallabMask::convertParamToNormal()
+{
+    const LocallabParams::LocallabSpot defSpot;
+
+    // Disable all listeners
+    disableListener();
+
+
+    lapmask->setValue(defSpot.lapmask);
+    gammask->setValue(defSpot.gammask);
+    slopmask->setValue(defSpot.slopmask);
+    shadmask->setValue(defSpot.shadmask);
+    str_mask->setValue(defSpot.str_mask);
+    ang_mask->setValue(defSpot.ang_mask);
+    strumaskmask->setValue(defSpot.strumaskmask);
+    toolmask->set_active(defSpot.toolmask);
+    fftmask->set_active(defSpot.fftmask);
+    blurmask->setValue(defSpot.blurmask);
+    contmask->setValue(defSpot.contmask);
+    HHhmask_shape->setCurve(defSpot.HHhmask_curve);
+//    Lmask_shape->setCurve(defSpot.Lmask_curve);
+    LLmask_shapewav->setCurve(defSpot.LLmask_curvewav);
+    csThresholdmask->setValue<int>(defSpot.csthresholdmask);
+
+    // Enable all listeners
+    enableListener();
+
```

Comment:
@Desmis 
`updatemaskGUI3` function shall be called here as `fftmask` has been set to a default value.

Pierre